orders/tests.py

```diff
@@ -52,3 +52,55 @@ def test_dog_owner_id_is_deleted_when_dog_owner_is_deleted(self, create_order):
         assert Order.objects.get(id=create_order.id).daycare_id is None
+
+    @pytest.mark.parametrize("dog_owner_1_id, dog_owner_2_id, delta_1, delta_2, capacity, expected_result",
+                             [(1, 2, 3, 5, 1, False),
+                              (2, 5, 4, 3, 10, True),
+                              (5, 4, 6, 7, 15, True),
+                              (4, 3, 4, 2, 1, False)])
+    def test_order_is_approvable_according_to_daycare_capacity(self, dog_owner_1_id, dog_owner_2_id, delta_1, delta_2,
+                                                               capacity, expected_result):
+        daycare = DayCare.create("capacity_email@gmail.com", "CapacityUserName", "CapacityPassword", "CapacityName",
+                                 "Changeable capacity", 100, capacity, "Merkaz", "Tel Aviv", "Capacity 123")
+        dog_owner_1 = DogOwner.objects.get(id=dog_owner_1_id)
+        dog_owner_2 = DogOwner.objects.get(id=dog_owner_2_id)
+
+        order1 = Order.create(start_date=timezone.now(), end_date=timezone.now() + datetime.timedelta(days=delta_1),
+                              daycare_id=daycare, dog_owner_id=dog_owner_1, price_per_day=100)
+        order2 = Order.create(start_date=timezone.now(), end_date=timezone.now() + datetime.timedelta(days=delta_2),
+                              daycare_id=daycare, dog_owner_id=dog_owner_2, price_per_day=100)
+
+        order1.approve_order()
+        assert order2.is_the_order_approvable() == expected_result
+
+    @pytest.mark.parametrize("new_status, expected_result", [(StatusOptions.Pending, True),
+                                                             (StatusOptions.Canceled, False),
+                                                             (StatusOptions.Approved, True),
+                                                             (StatusOptions.OnGoing, False),
+                                                             (StatusOptions.Finished, False)])
+    def test_order_cancellation_according_to_order_status(self, create_order, new_status, expected_result):
+        create_order.status = new_status
+        assert create_order.is_the_order_cancelable() == expected_result
+
+    @pytest.mark.parametrize("new_status, expected_result", [(StatusOptions.Pending, "Pending"),
+                                                             (StatusOptions.Canceled, "Canceled"),
+                                                             (StatusOptions.Approved, "Approved"),
+                                                             (StatusOptions.OnGoing, "Ongoing"),
+                                                             (StatusOptions.Finished, "Finished")])
+    def test_get_order_status_as_string(self, create_order, new_status, expected_result):
+        create_order.status = new_status
+        assert create_order.get_order_status() == expected_result
+
+    def test_get_daycare_capacity_in_dates_range(self, create_order):
```

Comment:
That's not how you should test this function. it has a lot of logic; you need to test different cases and see that you get the capacity you expect in each case

Reply:
I thought that checking the current capacity of a daycare before and after approving an order will check if the method is working correctly.
Also, some of the other tests are testing this method too (like "test_order_is_approvable_according_to_daycare_capacity").

Do you have any example for a case that I should test for this method? Or another way for testing it?

Reply:
second:
the test above test if order can be approved, according to its overall calculated capacity
this test should be a specific unit-test for `get_capacity_of_daycare_in_dates_range` it doesn't test the overall logic your app will go through, but it is testing this method specifically, and should pass even if in the future you'll decide to change the way the process works. 

this method has a lot of cases. think of each `if` statement as at least two possible test-cases.

Reply:
@OfirMatasas are you going to address these?